hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -874,7 +874,7 @@ public CompletableFuture<Message> applyTransaction(TransactionContext trx) {
       // if waitOnBothFollower is false, remove the entry from the cache
       // as soon as its applied and such entry exists in the cache.
       if (!waitOnBothFollowers) {
-        stateMachineDataCache.removeIf(k -> k >= index);
+        stateMachineDataCache.removeIf(k -> k <= index);
```

Comment:
Nit: Can you please extract this to a new method to emphasize the difference from `truncate`?  It would also help avoid calculating `Math.min(minIndex, index)` for each item at the other call.

Reply:
@adoroszlai Updated...

Reply:
Thanks.  Sorry if it was not clear, but I meant extracting the `stateMachineDataCache.removeIf(k -> k <= index);` line to a method that can then be called from both places where `<=` condition is used.